lhotse/cut/set.py

```diff
@@ -2638,127 +2658,14 @@ def _takewhile(
     return collected, iterable
 
 
-def merge_supervisions(
```

Comment:
Do I get it right -- you moved `merge_supervisions` to each of the cut classes definition because it's cleaner to implement them separately. In that case, can you keep `merge_supervisions` for CutSet that simply returns `self.map(lambda cut: cut.merge_supervisions(...))`?

Reply:
Yeah, that's right. Will do.

Reply:
Actually, the `CutSet` already has merge_supervisions(). I only removed the global utility function which implemented the merging logic and moved the implementation to each Cut type.